src/StackExchange.Redis/AzureMaintenanceEvent.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using static StackExchange.Redis.ConnectionMultiplexer;
+
+namespace StackExchange.Redis
+{
+    /// <summary>
+    /// Azure node maintenance event. For more information, please see: https://github.com/Azure/AzureCacheForRedis/blob/main/AzureRedisEvents.md
+    /// </summary>
+    public class AzureMaintenanceEvent
+    {
+        private const string PubSubChannelName = "AzureRedisEvents";
+
+        internal AzureMaintenanceEvent(string azureEvent)
+        {
+            if (azureEvent == null)
+            {
+                return;
+            }
+
+            // The message consists of key-value pairs delimted by pipes. For example, a message might look like:
+            // NotificationType|NodeMaintenanceStarting|StartTimeUtc|2021-09-23T12:34:19|IsReplica|False|IpAddress|13.67.42.199|SSLPort|15001|NonSSLPort|13001
+            var message = new ReadOnlySpan<char>(azureEvent.ToCharArray());
+            try
+            {
+                while (message.Length > 0)
+                {
+                    if (message[0] == '|')
+                    {
+                        message = message.Slice(1);
+                        continue;
+                    }
+
+                    // Grab the next pair
+                    var key = message.Slice(0, message.IndexOf('|'));
+                    message = message.Slice(key.Length + 1);
+
+                    var valueEnd = message.IndexOf('|');
+                    var value = valueEnd > -1 ? message.Slice(0, valueEnd) : message;
+                    message = message.Slice(value.Length);
+
+                    if (key.Length > 0 && value.Length > 0)
+                    {
+                        switch (key)
+                        {
+                            case var _ when key.SequenceEqual(nameof(NotificationType).ToCharArray()):
+                                NotificationType = value.ToString();
+                                break;
+                            case var _ when key.SequenceEqual(nameof(StartTimeInUTC).ToCharArray()) && DateTime.TryParse(value.ToString(), out DateTime startTime):
```

Comment:
These `.ToString()` are avoidable; on modern TFMs, there are span overloads on the types directly, but before that: consider `System.Buffers.Text.Utf8Parser.TryParse`, which has a range of APIs; also - and this is a very important one: think i18n/l10n: I *would* say "explicitly specify invariant", but I believe if using `Utf8Parser`: it is assumed - you should probably check on that, though

Reply:
I tried applying the Utf8Parser, but it deals with ReadOnlySpan<byte> instead of ReadOnlySpan<char>, which is what we're working with here. Adding the extra conversion between the two seems like the opposite of what we're trying to do here? Also [the documentation](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.text.utf8parser.tryparse?view=net-5.0) doesn't mention anything about cultures, so I think it makes sense to opt for the DateTime.TryParseExact() option for StartTimeUtc so we can specify it explicitly, as per your other comment. 
I ended up using the preprocessor directive you mentioned in another comment to get rid of all the .ToString() conversions for the newer frameworks.

Reply:
> The DateTime parse is of particular concern to me; I wonder if the format should be explicit/"exact" here (i.e. `DateTime.TryParseExact`) - or is it sufficient to use the `'u'` pattern (which is not the default) on `Utf8Parser`? Does `'u'` act the same [as `"u"`?](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#UniversalSortable) (open question; I don't know the answer)

Regarding this comment -- on the Azure side we're sending these events with the "s" (sortable) format, but according to VS 'u' looks like this: 

![image](https://user-images.githubusercontent.com/13313748/135896938-ee96daff-ab83-4076-85b0-84bbb7b8ab1e.png)

And 'U' looks like: 

![image](https://user-images.githubusercontent.com/13313748/135896952-26905e5f-1c29-4c96-aae4-aed577e1e852.png)

Reply:
fair enough; whatever works :) My key point: we need to make sure it is culture-insensitive, and we should *probably* validate that in a test by changing (and restoring in  a `finally`) the thread's culture and ui-culture to something random and suitably awkward